src/pyodide/internal/python.ts

```diff
@@ -38,7 +39,16 @@ import { TRANSITIVE_REQUIREMENTS } from 'pyodide-internal:metadata';
  * `noInitialRun: true` and so the C runtime is in an incoherent state until we
  * restore the linear memory from the snapshot.
  */
-function prepareWasmLinearMemory(Module: Module): void {
+function prepareWasmLinearMemory(
+  Module: Module,
+  pyodide_entrypoint_helper: PyodideEntrypointHelper | null
+): void {
+  if (!pyodide_entrypoint_helper) {
+    throw new PythonRuntimeError(
+      'pyodide_entrypoint_helper is required to load snapshot'
+    );
+  }
```

Comment:
Does it not typecheck if we do:
```suggestion
  pyodide_entrypoint_helper: PyodideEntrypointHelper
): void {
```